techmatters/webchat
# End chat triggers post survey when the task is in assigned state

## Description
- This PR along with this serverless PR [#340](https://github.com/techmatters/serverless/pull/340) allows user see and fill out the post survey when they end the chat
- Refactored End chat button to call `postSurveyInit` function and use `taskSid` and `isTaskStageAssigned` that are part of endChat response body and for control flow.

### Checklist

### Related Issues
Fixes #[CHI-1402](https://bugs.benetech.org/browse/CHI-1402)

### Verification steps
- Ensure that this [PR](https://github.com/techmatters/serverless/pull/340) for serverless is deployed to Aselo production.
- Try 'End Chat' when the task is in the following states: pending, reserved, assigned. In the case of assigned, make sure there is opportunity to start post survey
